src/limo/api.clj

```diff
@@ -63,10 +61,16 @@
   [s]
   (cond
     (element? s) (By/id (.getId s))
-    (:xpath s) (By$ByXPath. (:xpath s))
+    (:xpath s) (By/xpath (:xpath s))
     (:id s) (By/id (:id s))
-    (:css s) (By$ByCssSelector. (:css s))
-    :else (By$ByCssSelector. s)))
+    (:tag-name s) (By/tagName (:tag-name s))
+    (:link-text s) (By/linkText (:link-text s))
+    (:partial-link-text s) (By/partialLinkText (:partial-link-text s))
+    (:name s) (By/name (:name s))
+    (:class-name s) (By/className (:class-name s))
+    (or (:css s) ;; for backward compatibility
+        (:css-selector s)) (By/cssSelector (:css-selector s))
```

Comment:
@agilecreativity Maybe I'm missing something, but for the backward compatibility to work here, don't we  need the `(or (:css s) (:css-selector s))` in both the condition and in `By/cssSelector`? It might be easier to just keep `:css` and `:css-selector` as two different conditions in the `cond`.

I think since we didn't have fine grained tests in this section (as you've now added), it wasn't caught.